src/core.c/List.pm6

```diff
@@ -190,6 +190,21 @@ my class List does Iterable does Positional { # declared in BOOTSTRAP
         nqp::p6bindattrinvres(result,List,'$!todo',todo)
     }
 
+    method over-iterator(List:D: Iterator $iter --> List:D) {
+        # from-iterator only copies type-level information; here we copy
+        # everything bar the elements (e.g. container descriptors, shapes)
+        nqp::stmts(
```

Comment:
Why the nqp::stmts??

Reply:
I've seen it produce a simpler QAST before, but I don't think I checked `--optimize`'s output at the time.

Reply:
That may be true, but for readability, it would definitely improve if `nqp::stmts` were not used when not really needed for performance.  And I don't think it is needed here.  And I doubt you would be able to show any difference in execution once spesh and inlining done their work.

Yes, there used to be a time when it did make a significant difference.  But that became a lot less when Parrot was not the backend anymore.  And with spesh it became even less.

Reply:
I don't depend on `nqp::stmts` here anymore, but still do in the context of `StructuredBuffer`'s methods, which are depended on from the context of an `Iterator`. `Gauge`'s iterators are sensitive to it vs. actual statements, so I imagine a list assignment would be too.